### PR TITLE
Add docstrings and examples

### PR DESCRIPTION
**Pull request by sushobhana**
as discussed in #194 
DS9:
region_ds9
regions + mpl:
region_mpl

### CODE REVIEW

**Comment by sushobhana** [comment]
@keflavich added docsting examples here.

**Comment by keflavich** [comment]
Nice work, the figures look great in the built docs

**Comment by sushobhana** [comment]
@keflavich Thanks! One thing is that there are some `methods`  or `attributes`  that inherit docstrings from it's abstract method but sphinx does not pick those up.  Any workaround for this?

**Comment by keflavich** [comment]
Yikes, I don't know about that.  Ask on the Slack channel?

**Comment by larrybradley** [comment]
For methods, you can try http://docs.astropy.org/en/stable/api/astropy.utils.misc.InheritDocstrings.html

**Comment by bsipocz** [comment]
@sushobhana - we also have some utilities in astroquery to append and prepend to the docstring, if that would be any use to you here.
Also at some point I planned to move that upstream to astropy in https://github.com/astropy/astropy/pull/6788, and maybe it will indeed happen for 3.1

**Comment by sushobhana** [comment]
Thanks @larrybradley @bsipocz . 
The only difference between `MetaRegion`( that I added here) and `InheritDocStrings` is that the former inherits `abc.ABCMeta`.

File: docs/plot_reg.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+ax = plt.gca()
+plt.imshow(image_data, cmap='gray')
+
+regs = []
+print(ax.get_xlim(), ax.get_ylim())
+ax.set_ylim([-0.5, 892.5])
+colors = ['red', 'blue', 'orange', 'brown', 'yellow', 'violet']
+
+regs.append(CirclePixelRegion(PixCoord(500, 600), 50))
```
[the remark this answers is not in the document]
Comment (sushobhana) [reply docs/plot_reg.py]: put all the regions in `plot_image.reg` file.

File: docs/index.rst [hunk 2]
```diff
@@ -39,6 +39,16 @@ User guide
    crtf
    shapely
 
+These plot are made using :download:`plot_image.reg`.
```
Comment (migueldvb) [inline docs/index.rst]: These plots

**Comment by sushobhana** [comment]
@migueldvb Thanks ! made the changes.

**Pull request merged**